components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -33,6 +36,29 @@ class GeckoEngineSession(
         geckoSession.trackingProtectionDelegate = createTrackingProtectionDelegate()
     }
 
+    /**
+     * See [EngineSession.settings]
+     */
+    override val settings: Settings = object : Settings, PrefsHelper.PrefHandlerBase() {
+        val prefs = mutableMapOf<String, Any>()
+        init { PrefsHelper.getPref(DOM_STORAGE_PREF, this) }
+
+        override var javascriptEnabled: Boolean
+            get() = runtime.settings.javaScriptEnabled
```

Comment:
Oh, JavaScript is a runtime setting? GeckoView distinguishes between Runtime and Session settings. Should we do that too?

Reply:
Alright!